driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -16,9 +16,32 @@
 
 package com.mongodb.client.model.expressions;
 
+import static com.mongodb.client.model.expressions.Expressions.of;
+
 /**
  * Expresses a string value.
  */
 public interface StringExpression extends Expression {
 
+    StringExpression toLower();
+
+    StringExpression toUpper();
+
+    StringExpression concat(StringExpression concat);
```

Comment:
This appends the string to the current string. Should we have a prepend method? If so, should this method be renamed?

Reply:
When a question is posed this way, it forces others to think about it from scratch, then describe, pros/cons, potential usages, and suggest a decision. I think that all of this work should be done by the one who asks the question so that others could analyze the proposed options and express their opinion.

Reply:
Here is an example where prepend (first line) is much easier to naturally write, compared to the equivalent append:
```
[a, b, c, d].map(v -> v.toString().trim().first(3).prepend(">"))
[a, b, c, d].map(v -> of(">").append(v.toString().trim().first(3)))
```
We discussed this and are leaning towards using prepend/append here, though the reductive operation would be concat (see https://github.com/mongodb/mongo-java-driver/pull/1032#discussion_r1021929731 ). Additional input welcome.